akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -390,7 +426,7 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
   }
 
   private def writeLiteral(tagOffset: Int, literal: String): Unit = {
-    val length = literal.length
+    val length = if (literal eq null) 0 else literal.length
```

Comment:
We encode `null` and empty string as zero bytes, and decode it to empty string. For the `noSender` or `noRecipient` case we use `null` for the ref and -1 for the idx, i.e. the null is read as empty string when decoding it and therefore I turn it back to `null` for refs

empty string literal for manifest is valid, and then it should still be decoded as empty string